tests/components/nut/test_device_action.py

```diff
@@ -0,0 +1,201 @@
+    """Test actions are executed."""
+
+    list_commands_return_value = {
+        "beeper.enable": None,
+        "beeper.disable": None,
+    }
+    run_command = MagicMock()
+    await async_init_integration(
+        hass,
+        list_ups={"someUps": "Some UPS"},
+        list_vars={"ups.status": "OL"},
+        list_commands_return_value=list_commands_return_value,
+        run_command=run_command,
+    )
+    device_entry = next(device for device in device_registry.devices.values())
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        "platform": "event",
+                        "event_type": "test_some_event",
+                    },
+                    "action": {
+                        "domain": DOMAIN,
+                        "device_id": device_entry.id,
+                        "type": "beeper_enable",
+                    },
+                },
+                {
+                    "trigger": {
+                        "platform": "event",
+                        "event_type": "test_another_event",
+                    },
+                    "action": {
+                        "domain": DOMAIN,
+                        "device_id": device_entry.id,
+                        "type": "beeper_disable",
+                    },
+                },
+            ]
+        },
+    )
+
+    hass.bus.async_fire("test_some_event")
+    await hass.async_block_till_done()
+    run_command.assert_called_with("someUps", "beeper.enable")
+
+    hass.bus.async_fire("test_another_event")
+    await hass.async_block_till_done()
+    run_command.assert_called_with("someUps", "beeper.disable")
+
+
+async def test_rund_command_exception(
+    hass: HomeAssistant, device_registry: dr.DeviceRegistry
+) -> None:
+    """Test logged error if run command raises exception."""
+
+    list_commands_return_value = {"beeper.enable": None}
+    run_command = MagicMock(side_effect=PyNUTError)
+    await async_init_integration(
+        hass,
+        list_vars={"ups.status": "OL"},
+        list_commands_return_value=list_commands_return_value,
+        run_command=run_command,
+    )
+    device_entry = next(device for device in device_registry.devices.values())
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        "platform": "event",
+                        "event_type": "test_some_event",
+                    },
+                    "action": {
+                        "domain": DOMAIN,
+                        "device_id": device_entry.id,
+                        "type": "beeper_enable",
+                    },
+                },
+            ]
+        },
+    )
+
+    with mock.patch("logging.Logger.error") as error:
```

Comment:
We shouldn't patch the logger.  Instead if you want to assert something is logged, use the caplog fixture

Reply:
I still have pending fixing this one. I'll get back to it shortly.